feat: create suricata.md with basic info (#9)
naslabsec/Infrastructure: docs/SURICATA.md

# Suricata
Suricata is a high performance, open source network analysis and threat detection software

## Requirements
`apt install suricata`

## Configurations
With the following steps you can install Suricata and get the right config to start using it quick-start.

> 1. in File: `/etc/suricata/suricata.yaml` go to line 1862 and substitute `- suricata.rules` with `- /ctf/ipsrules/local.rules`

Reviewer: ```suggestion
1. In `/etc/suricata/suricata.yaml` go to line 1862 and substitute `- suricata.rules` with `- /ctf/ipsrules/local.rules`
```

2. Create rules file `/ctf/ipsrules/local.rules` add at least one rule [How to write rules](https://docs.suricata.io/en/suricata-6.0.0/rules/intro.html)

> 	#example: 

Reviewer: This is rendered correctly, remove the `Tab` and add a space between `#` and the title.
Even tho a title would look bad here, maybe put `###` or `####`

	```
	alert tcp $HOME_NET 1984 -> $EXTERNAL_NET any (msg: "Path Traversal-../"; content: "../"; metadata: tag path_traversal; sid:1; rev: 1;)
	alert tcp any any -> any any (msg: "Path Global-../"; flow:to_server; content: "../"; metadata: tag path_traversal; sid:2; rev: 2;)
	```

3. !IMPORTANT! Set-up **iptables rules** that sends pkts to NFQUEUE. Write the rules exactly like this with right IP and Port (add both input and output rules):
	```bash
	ip = "localhost"
	dp = "1337"
	sudo iptables -I INPUT -d "$ip" -p tcp --dport "$dp" -j NFQUEUE --queue-num 1 --queue-bypass
	sudo iptables -I OUTPUT -d "$ip" -p tcp --sport "$dp" -j NFQUEUE --queue-num 1 --queue-bypass
	```
	
1. Start Suricata in [IPS mode](https://suricata.readthedocs.io/en/suricata-6.0.0/setting-up-ipsinline-for-linux.html) using screen

> 	`sudo screen -dmS surica suricata -c /etc/suricata/suricata.yaml -q 0`	

Reviewer: ```suggestion
	```sudo screen -dmS surica suricata -c /etc/suricata/suricata.yaml -q 0```
```
A codeblock would be better here. Also, I would explain what the option of the command do.

## Usage

> ### start and restart

Reviewer: ```suggestion
### Start & Restart
```
Something like this would look visually better.

Start Suricata in [IPS mode](https://suricata.readthedocs.io/en/suricata-6.0.0/setting-up-ipsinline-for-linux.html) using screen:

> `sudo screen -dmS surica suricata -c /etc/suricata/suricata.yaml -q 0`	

Reviewer: Same thing as the previous screen command

Restart Suricata after modifying rules or config:

> `screen -r surica` and ctrl-c to close the process

Reviewer: ```suggestion
`screen -r surica` and Ctrl-C to close the process
```
The capital letters would be better

> ### rule writing

Reviewer: ```suggestion
### Writing Rules
```

> example

Reviewer: ```suggestion
#### Example
```

> `alert tcp $HOME_NET 1337 -> $EXTERNAL_NET 1337 (msg: "Path Traversal-../"; content: "../"; metadata: tag path_traversal; sid:1; rev: 1;)`

Reviewer: ```suggestion
```alert tcp $HOME_NET 1337 -> $EXTERNAL_NET 1337 (msg: "Path Traversal-../"; content: "../"; metadata: tag path_traversal; sid:1; rev: 1;)```
```
A codeblock here would look better

# Suricata
Suricata is a high performance, open source network analysis and threat detection software

## Requirements
`apt install suricata`

## Configurations
With the following steps you can install Suricata and get the right config to start using it quick-start.

1. in File: `/etc/suricata/suricata.yaml` go to line 1862 and substitute `- suricata.rules` with `- /ctf/ipsrules/local.rules`

2. Create rules file `/ctf/ipsrules/local.rules` add at least one rule [How to write rules](https://docs.suricata.io/en/suricata-6.0.0/rules/intro.html)
	```
	alert tcp $HOME_NET 1984 -> $EXTERNAL_NET any (msg: "Path Traversal-../"; content: "../"; metadata: tag path_traversal; sid:1; rev: 1;)
	alert tcp any any -> any any (msg: "Path Global-../"; flow:to_server; content: "../"; metadata: tag path_traversal; sid:2; rev: 2;)
	```

> 3. !IMPORTANT! Set-up **iptables rules** that sends pkts to NFQUEUE. Write the rules exactly like this with right IP and Port (add both input and output rules):

Reviewer: ```suggestion
3. **!IMPORTANT!** Setup **iptables rules** that sends `pkts` to `NFQUEUE`. Write the rules **exactly** like this with right IP and port (add both input and output rules):
```

	```bash
	ip = "localhost"
	dp = "1337"
	sudo iptables -I INPUT -d "$ip" -p tcp --dport "$dp" -j NFQUEUE --queue-num 1 --queue-bypass
	sudo iptables -I OUTPUT -d "$ip" -p tcp --sport "$dp" -j NFQUEUE --queue-num 1 --queue-bypass
	```
	

> 4. Start Suricata in [IPS mode](https://suricata.readthedocs.io/en/suricata-6.0.0/setting-up-ipsinline-for-linux.html) using screen

Reviewer: ```suggestion
```

Reviewer: Also, why is this in the `Configuration` section and then repeated in the `Usage` section? Just remove it from here.
